packages/profiles/test/musig/btc.musig.registration.test.ts

```diff
@@ -0,0 +1,292 @@
+			// .get("/api/peers")
+			// .reply(200, require("../fixtures/client/peers.json"))
+			// .get("/api/node/configuration/crypto")
+			// .reply(200, require("../fixtures/client/cryptoConfiguration.json"))
+			// .get("/api/node/syncing")
+			// .reply(200, require("../fixtures/client/syncing.json"))
+			// // Wallet mocks
+			// .get("/api/wallets/DABCrsfEqhtdzmBrE2AU5NNmdUFCGXKEkr")
+			// .reply(200, require("../fixtures/wallets/DABCrsfEqhtdzmBrE2AU5NNmdUFCGXKEkr.json"))
+			// .get("/api/wallets/DCX2kvwgL2mrd9GjyYAbfXLGGXWwgN3Px7")
+			// .reply(200, require("../fixtures/wallets/DCX2kvwgL2mrd9GjyYAbfXLGGXWwgN3Px7.json"))
+			// .get("/api/wallets/DDHk393YcsxTPN1H5SWTcbjfnCRmF1iBR8")
+			// .reply(200, require("../fixtures/wallets/DDHk393YcsxTPN1H5SWTcbjfnCRmF1iBR8.json"))
+			.persist();
+
+		profile1 = makeProfile({ id: "profile-id-1" });
+		profile2 = makeProfile({ id: "profile-id-2" });
+		profile3 = makeProfile({ id: "profile-id-3" });
+	});
+
+	afterEach(() => {
+		resetServerResponseMocks();
+	});
+
+	describe("MuSig Registration", () => {
+		it("should perform a 2 out of 2 registration", async () => {
+			// TODO We shouldn't need to create a wallet first, but DW requires so, as the button to create musig wallet is inside an existing wallet
+			const first = await bitcoinWallet({
+				profile: profile1,
+				coinId: "BTC",
+				networkId: "btc.testnet",
+				mnemonic: testMnemonics["btc.testnet"][0],
+				account: 0,
+				// path: "m/48'/1'/0'/2'",
+			});
+
+			const signatory1 = new Signatories.Signatory(
+				new Signatories.MnemonicSignatory({
+					signingKey: testMnemonics["btc.testnet"][0],
+					address: "address", // Not needed / used
+					publicKey: "m/48'/1'/0'/2'", // TODO for now we use publicKey for passing path
+					privateKey: "privateKey", // Not needed / used
+				}),
+			);
+
+			const transactionData = {
+				signatory: signatory1,
+				data: {
+					min: 2,
+					numberOfSignatures: 2,
+					publicKeys: [],
+					derivationMethod: "nativeSegwitMusig",
+				},
+			};
+
+			// 1. Create and sign musig registration by first
+			const uuid = await first
+				.transaction()
+				// @ts-ignore TODO Not sure why it doesn't like the type as it is
+				.signMultiSignature(transactionData);
+			console.log(uuid);
+
+			// Broadcast the transaction to musig server with first's signature.
+			// All participant wallets should see when calling wallet.transaction.sync()
+			mockServerResponse("store", { id: uuid });
+			const result = await first.transaction().broadcast(uuid);
+
+			expect(result).toEqual({ accepted: [uuid], errors: {}, rejected: [] });
+
+			// Validate multi-signature registration data.
+			expect(first.transaction().canBeBroadcasted(uuid)).toBeFalse();
+			// expect(first.transaction().canBeSigned(uuid)).toBeFalse(); // TODO returning true because profiles is checking against the pub key of the originator's wallet (bip84 m/84'/1'/0') instead of the key that's actually used for musig (m/48'/1'/0'/2')
```

Comment:
here

Reply:
Is only the check against the public key instead of extended public key an issue or anything else? Is this something we could handle internally with flags to determine a coins MuSig scheme/method and then decide what to do to return the appropriate values? ARK and LSK for example have the MuSig types `standard` and `advanced`, could introduce something like that for BTC.

Reply:
@marianogoldman if I am not missing anything, what would be the secure way to check if an address or a public key or another suitable identifier has added its signature in the transactions signatures array)?

I this example, the wallet has the following btc address & public key
```
publicKey = tpubDDGE6CVZd7KfZbEA8YNYTA6iepUVd78Vi3WRia1hoZ6fqQBbHuJNRSdoTXet8KRnpxFauRSCtXfqWAMSoESG1kbiXZXdw9cA2JitSL1YbxL
address = tb1qcn3gm6p8nkc8ks7e9dv2uw89jtgy5p4s0749ny
```
and the signatures array for the transaction looks like 
```ts
        signatures: [
          'H3/sR9TyzeZuWYMfO5IlGOE2KV+aAUNI44SBoVRs5kwMRrJ/Y4jyJeZPP4ofaciswi4G4XfWPn6eBOTvbMJYZ48='
        ]

```
isn't there a way to cryptographically validate the signature against the public key or extended public key or address?

In profiles musig transaction service, most of the musig statuses depend on 
```ts
wallet.coin().multiSignature().needsWalletSignature(transaction(id).data(), publicKey);
```
so if we would be able to determine and validate if a public key or another identifier has added it's signature that would be enough to handle most of the rest of the internal statuses.

Reply:
@goga-m The problem with that example is that the original wallet is checking whether it's own public key is required in the signing. But that public key will never be part of the musig wallet, so it will always say that it is required.
For the original wallet to be able to determine if it can sign or not, it doesn't have to use its own public/private keys. It needs to derive new ones based on the derivation schema and path that were used for creating the musig wallet.
So then... the non-musig wallet (the originator one) needs to store additional data regarding the musig wallets it has created in order to check for pending transactions.